### ISSUE TITLE
Work Item source option is not available with the Azure devops extension

### ISSUE THREAD

**Issue by ppRt915** [OctopusDeploy/OctoTFS#242]
Hi Team,

We are using Azure DevOps to build a code from BitBucket repository, and for deployment we are using Octopus Deploy. The Octopus deploy extension is already installed. Earlier I used it to deploy and used Azure devops work items as issue tracker. This time I am trying to use Jira and I am unable to do as work items source option is not available for me for task "Push build Information". I noticed The extension has a new version, and its already installed. Could you please help me with this.

![image](https://user-images.githubusercontent.com/105238748/167472228-8a8abe7c-f804-4322-8388-f7f89fadc38f.png)

**Comment by jbristowe** [comment]
Hello, @ppRt915! :wave: Thanks for submitting an issue to our repo.

I'm struggling to understand your issue. You mention that you're unable to use Jira as "work items source option".  Yet, this repository is for our integration with Azure DevOps. Are you seeking help with our Jira integration?

**Comment by slewis74** [comment]
Hi @ppRt915 ! I think I know what is happening, and if I'm correct I do apologise that this is a bit confusing. Let me see if I can help clarify.
The way the integration works is that the Build Information step (the one from your screenshot) passes information about the build and the commits to Octopus. 
When you view the package details, or create a release, In Octopus using the package/versions you specified on the step it will look at which [issue tracker](https://octopus.com/docs/releases/issue-tracking) extensions you have enabled and ask the enabled ones whether they recognise any work item data in the build information. If you have the Jira integration enabled and it sees what it thinks are work items references it will confirm them with Jira, and if they're valid they become visible in the UI and available for things like release notes etc.
You'd enable the Azure DevOps extension in Octopus if you want it to reach back and see which work items where attached to the build in ADO (it doesn't use commit parsing for it's processing).
So to recap, the step in Azure DevOps is simple about getting Build Information in to Octopus, it doesn't need to set a "work item source option". From there the results depend on the issue tracker extensions in Octopus itself. So to get what I think you're after, just set the Build Information step up in AzureDevOps and then enable the Jira integration in Octopus. Hope that helps.

**Comment by jbristowe** [comment]
Closing due to inactivity.
